EXT_BeamPhysics.md

```diff
@@ -0,0 +1,269 @@
+
+- `time-refTime`
+- Type: Optional *(float)*
+- Description: Particle time minus the referece time.
+
+- `energy/`
+  - Type: Optional *(float)*
+  - Description: The total energy of the particles relative to the `referenceEnergy` attribute.
+
+- `electricField/`
+    - Type: Optional 2-component *(float)*
+    - Description: Electric field. Used for photons only.
+    - Components: (`x`, `y`).
+        - For each component, the field is specified using either (`amplitude`, `phase`) or (`Real`, `Imaginary`)
+        subcomponents.
+
+- `macroCharge/`
+    - Type: Optional *(float)*
+    - Description: "Macro"-particles are simulation particles that represent multiple real particles.
+    The `macroCharge` is the the total charge of the collection of particles that a macro-particle
+    represents.
+
+- `momentum/`
+    - Type: Optional 2 or 3-vector *(float)*
+    - Description: The total momentum of the particles relative to the `momentumOrigin` attribute.
+    - Components: (`px`, `py`) or (`px`, `py`, `pz`).
+    - Note: A program using phase space coordinates to describe particle positions will typically use the transverse momentum
+    `px` and `py` along with the `totalMomentum` or `energy` records.
+
+- `pathLength/`
+    - Type: Optional *(float)*
+    - Description: Length that a particle has traveled.
+
+- `position`
+    - Type: Required 3-vector *(float)*
+    - Description: Position relative to the coordinate origin.
+    - Components: (`x`, `y`, `<z-coord>`) where `<z-coord>` is one of:
+        - `z`: True longitudinal coordinate.
+        - `-beta.c.dt`: Phase space coordinate conjugate to a momentum based "pz".
+        - `-beta.c.t`:Phase space coordinate conjugate to a momentum based "pz".
+        - `-c.dt`: Phase space coordinate conjugate to an energy based "pz".
+        - `-c.t`: Phase space coordinate conjugate to an energy based "pz".
+        - Where: **beta** = particle speed, **c** = speed of light, **t** = time, and **dt** = time relative to the reference time.
+
+- `refTime/`
+    - Type: Optional *(float)*
+    - Description: The reference particle time.
+    Note that the reference time is a function of **s** and therefore can be different for different particles.
+
+- `s-Position`
+    - Type: Optional *(float)*
+    - Description: Absolute longitudinal distance in **lattice** coordinates.  
+
+- `speed/`
+    - Type: Optional *(float)*
+    - Description: The speed of the particles.
+
+- `spin/`
+    - Type: Optional 3-vector *(float)*
+    - Description: Particle spin.
+    - Components: (`x`, `y`, `s`) or (`r`, `theta`, `phi`).
+
+- `time/`
+    - Type: Optional *(float)*
+    - Description: Absolute particle time. Note: Particles may have different times if the snapshot
+    is, for example, taken at constant **s**.
+
+- `totalMomentum/`
+    - Type: Optional *(float)*
+    - Description: The total momentum of the particles relative to the `referenceTotalMomentum` attribute.
```

Comment:
Just for my own understaning, does this mean `sqrt(px**2 + py**2 + pz**2)`?
Also, the same remarks as for `energy` apply (regarding normalization and redundancy with `unitSI`).

Reply:
@RemiLehe 

> Just for my own understanding, does this mean sqrt(px**2 + py**2 + pz**2)?

Yes if px, py, pz are unnormalized momentum coordinates. Typically, px, and py used by accelerator programs are normalized momentum coordinates and pz could be something like energy deviation and not at all directly related to the longitudinal momentum.

Reply:
@DavidSagan the phrase "relative to" could mean an additive factor, or a multiplicative factor. Where do we define what this means?

Reply:
@ChristopherMayes: In the upcoming revision I have:

- Attributes: The following attributes are defined.
      - `normalized`:
        - Type: Required **(bool)**
        - Description: Is the energy normalized by the `referenceTotalMomentum`? That is, is `px` equal to Px or Px/P0 (where Px is the true momentum component and P0 is `referenceTotalMomentum`)?
    - Note: A program using phase space coordinates to describe particle positions will typically use the transverse momentum `px` and `py` along with the `totalMomentum` or `energy` records.